packages/material-ui/src/ExpansionPanel/ExpansionPanel.js

```diff
@@ -78,93 +78,87 @@ export const styles = theme => {
   };
 };
 
-class ExpansionPanel extends React.Component {
-  constructor(props) {
-    super();
-    this.isControlled = props.expanded != null;
-    this.state = {};
-    if (!this.isControlled) {
-      // not controlled, use internal state
-      this.state.expanded = props.defaultExpanded !== undefined ? props.defaultExpanded : false;
+function useIsControlled(value) {
```

Comment:
What's the value of this abstraction?

Reply:
This was me playing around a bit. I have a more generic abstraction locally that will deal with the controlled vs. uncontrolled value in general. 

Though even in this primitive state it adds value: We can move possible discussions about what constitutes a "controlled" value into a single place (see https://github.com/mui-org/material-ui/pull/14092#discussion_r245480818). This would ensure a consistent controlled vs. uncontrolled API.